controllers/components/workbenches/workbenches_controller_actions.go

```diff
@@ -0,0 +1,120 @@
+package workbenches
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	componentsv1 "github.com/opendatahub-io/opendatahub-operator/v2/apis/components/v1"
+	odhtypes "github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/types"
+	odhdeploy "github.com/opendatahub-io/opendatahub-operator/v2/pkg/deploy"
+)
+
+func initialize(ctx context.Context, rr *odhtypes.ReconciliationRequest) error {
+	rr.Manifests = append(rr.Manifests, odhtypes.ManifestInfo{
+		Path:       NotebookControllerPath,
+		ContextDir: "",
+		SourcePath: "",
+	})
+	if err := odhdeploy.ApplyParams(
+		NotebookControllerPath,
+		nil,
+		map[string]string{
+			"namespace": rr.DSCI.Spec.ApplicationsNamespace,
+		},
+	); err != nil {
+		return fmt.Errorf("failed to update params.env from %s : %w", rr.Manifests[0], err)
+	}
+
+	rr.Manifests = append(rr.Manifests, odhtypes.ManifestInfo{
+		Path:       KfNotebookControllerPath,
+		ContextDir: "",
+		SourcePath: "",
+	})
+	if err := odhdeploy.ApplyParams(
+		KfNotebookControllerPath,
+		nil,
+		map[string]string{
+			"namespace": rr.DSCI.Spec.ApplicationsNamespace,
+		},
+	); err != nil {
+		return fmt.Errorf("failed to update params.env from %s : %w", rr.Manifests[1], err)
+	}
+
+	rr.Manifests = append(rr.Manifests, odhtypes.ManifestInfo{
+		Path:       NotebookImagesPath,
+		ContextDir: "",
+		SourcePath: "",
+	})
+	if err := odhdeploy.ApplyParams(
+		NotebookImagesPath,
+		nil,
+		map[string]string{
+			"namespace": rr.DSCI.Spec.ApplicationsNamespace,
+		},
+	); err != nil {
+		return fmt.Errorf("failed to update params.env from %s : %w", rr.Manifests[2], err)
+	}
+
+	return nil
+}
+
+func devFlags(ctx context.Context, rr *odhtypes.ReconciliationRequest) error {
+	workbenches, ok := rr.Instance.(*componentsv1.Workbenches)
+	if !ok {
+		return fmt.Errorf("resource instance %v is not a componentsv1.Workbenches)", rr.Instance)
+	}
+
+	if workbenches.Spec.DevFlags == nil {
+		return nil
+	}
+
+	// Download manifests if defined by devflags
+	// Go through each manifest and set the overlays if defined
+	// first on odh-notebook-controller and kf-notebook-controller last to notebook-images
+	if len(workbenches.Spec.DevFlags.Manifests) != 0 {
+		for _, subcomponent := range workbenches.Spec.DevFlags.Manifests {
```

Comment:
in this loop, you should amend the rr.Manifesst path, otherwhise the changes won't be taken into account, simlat to https://github.com/opendatahub-io/opendatahub-operator/blob/feature-operator-refactor/controllers/components/dashboard/dashboard_controller_actions.go#L49-L59